clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -214,7 +212,8 @@ private KeyStore load() throws GeneralSecurityException, IOException {
             try {
                 KeyStore ks = KeyStore.getInstance(type);
                 in = new FileInputStream(path);
-                ks.load(in, password.value().toCharArray());
+                char[] passwordChars = password != null ? password.value().toCharArray() : null;
```

Comment:
Worth adding a comment explaining that a null password means that integrity checking is disabled, but access to the truststore is still available.

We should probably also update the documentation for the config (in `SslConfigs`).